### ISSUE TITLE
[Nestjs][ExpressAdapter][Bull] HttpAdapterHost isn't available in the BullBoardRootModule context

### ISSUE THREAD

**Issue by TrueGelen** [felixmosh/bull-board#590]
I get an error when building the project in a dev mode.

### The following error:
Nest can't resolve dependencies of the BullBoardRootModule (?, bull_board_adapter, bull_board_options). Please make sure that the argument HttpAdapterHost at index [0] is available in the BullBoardRootModule context.

Potential solutions:
- If HttpAdapterHost is a provider, is it part of the current BullBoardRootModule?
- If HttpAdapterHost is exported from a separate @Module, is that module imported within BullBoardRootModule?
  @Module({
    imports: [ /* the Module containing HttpAdapterHost */ ]
  })
  
### some part of packahe.json
```
    "@bull-board/api": "^5.5.1",
    "@bull-board/express": "^5.5.1",
    "@bull-board/nestjs": "^5.5.1",
    "@nestjs/bull": "^0.6.0",
    "@nestjs/core": "^9.0.5",
```
    
### some part of app.module
```
@Module({
  imports: [
    QueueModule,

    BullBoardModule.forRoot({
      route: '/queues',
      adapter: ExpressAdapter,
    }),
  ],
})
```

### queue.module.module
```
@Module({
  imports: [
    BullModule.forRoot({
      redis: {
        host: process.env.REDIS_HOST || 'localhost',
        port: Number(process.env.REDIS_PORT) || 6379,
      },
    })
  ],
  exports: [QueueService],
  providers: [QueueService],
})
export class QueueModule {
  static registerQueue(options): DynamicModule {
    return BullModule.registerQueue(options);
  }
}
```

**Comment by felixmosh** [comment]
Hi,  thank you for reporting this issue.
Maybe @DennisSnijder can help?

**Comment by DennisSnijder** [comment]
@TrueGelen I can't seem to recreate the problem with the packages you mentioned. Can you create a test repo with a reproducible example? 

The dependency it's trying to inject (which is missing in your case) is a built in service provided by NestJS.
Are you perhaps using NestJS microservices? I can imagine NestJS microservices might not have the HttpAdapter built in.

**Comment by russosalv** [comment]
i have same error

mine main.ts is that one
```typescript
import { NestFactory } from '@nestjs/core';
import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
import * as appInsights from 'applicationinsights';
import { AppModule } from './app.module';
import { ApplicationInsightInterceptor } from './applicationInsight/applicationInsight.interceptor';
import { ApplicationInsightUtils } from './applicationInsight/applicationInsight.utils';
import { ApplicationInsightLogger } from './applicationInsight/applicationInsightLogger.utils';
import { AuxiliaryJobService } from './modules/auxiliary-job/auxiliary-job-service';
import { GetterJobService } from './modules/getter-job/getter-job.service';
import { SapSyncProcessConfigUtils } from './utils/sapSyncProcessConfig.utils';
import {
	AllExceptionFilter,
	SapRfcNoDataExtractedExceptionFilter,
	TooMuchRunNowJobFilter
} from "./filter/allException.filter";

async function bootstrap() {
	const app = await NestFactory.create(AppModule);

	console.log('DATABASE_URL', process.env.DATABASE_URL);
	console.log('REDIS_HOST', process.env.REDIS_HOST);
	console.log('REDIS_PORT', Number(process.env.REDIS_PORT));

	const applicationInsight = await app.resolve(ApplicationInsightUtils);
	app.useGlobalInterceptors(new ApplicationInsightInterceptor(applicationInsight));

	applicationInsight.defaultClient?.trackEvent({
		name: 'SapSyncV2 Started',
		properties: {
			startTimeUtc: new Date(),
		},
	} as appInsights.Contracts.EventTelemetry);

	const logger = await app.resolve(ApplicationInsightLogger);
	app.useLogger(logger);

	app.useGlobalFilters(new TooMuchRunNowJobFilter(applicationInsight));
	app.useGlobalFilters(new SapRfcNoDataExtractedExceptionFilter(applicationInsight));

	const swaggerConfig = new DocumentBuilder()
		.setTitle('syncV2')
		.setDescription('syncV2')
		.setVersion('2.0')
		.addBearerAuth({ type: 'http', scheme: 'bearer', bearerFormat: 'JWT' })
		.build();

	const document = SwaggerModule.createDocument(app, swaggerConfig, {
		// extraModels: [...PrismaModel.extraModels],
	});
	SwaggerModule.setup('api', app, document);

	const sapSyncProcessConfigUtils = await app.resolve(SapSyncProcessConfigUtils);
	const sapSyncProcessConfigs = await sapSyncProcessConfigUtils.getSapSyncProcessConfigsAll();
	const isFirstSyncAtAll = await sapSyncProcessConfigUtils.isFirstSyncAtAll();

	const getterJobService = await app.resolve(GetterJobService);
	const auxiliaryJobService = await app.resolve(AuxiliaryJobService);

	//** START BEFORE THE REGISTRATION OF QUEUES THEN DISCHARGE DATA */
	// if (isFirstSyncAtAll) {
	// 	await getterJobService.startFirstSync();
	// }
	// await auxiliaryJobService.rescueGetter();
	// await auxiliaryJobService.rescueSetter();
	// await auxiliaryJobService.registerDefault();
	// getterJobService.registerScheduled(sapSyncProcessConfigs);

	app.enableCors({
		origin: '*',
	});

	await app.listen(3000).then(async () => {

		//TODO: REMOVE THIS
		return;

		if (isFirstSyncAtAll) {
			await getterJobService.startFirstSync();
		}
		await auxiliaryJobService.rescueGetter();
		await auxiliaryJobService.rescueSetter();

		await auxiliaryJobService.registerDefault();
		getterJobService.registerScheduled(sapSyncProcessConfigs);
	});
}
bootstrap();
```

**Comment by foxsheldin** [comment]
@russosalv 
I just solved the problem in our project today and deleted the temporary solution @TrueGelen. 

As far as I understand, we have a problem at the dependency level of the project packages. We use a monorepository, and in our case the problem turned out to be an incorrect configuration of the Yarn package manager. 

I performed an operation to update `Yarn` version from 1.22.19 to version 4.1.0. When working with the configuration, I set the `HoistingLimits` property to `dependecies`, deleted all `node_modules` folders and reinstalled packages. Next, I fixed Typescript highlighted errors by installing or updating project dependencies. 

When updating the API server project, only 1 package was affected: @types/lodash. The rest of the dependencies, as it turned out, were taken from other projects of the monorepository
